src/NUnit.Xml.TestLogger/NUnitXmlTestLogger.cs

```diff
@@ -0,0 +1,501 @@
+        }
+
+        private static XElement CreateErrorElement(TestResultInfo result)
+        {
+            string errorMessage = result.ErrorMessage;
+
+            int indexOfErrorType = errorMessage.IndexOf('(');
+            errorMessage = errorMessage.Substring(indexOfErrorType + 1);
+
+            int indexOfName = errorMessage.IndexOf(')');
+            string name = errorMessage.Substring(0, indexOfName);
+            errorMessage = errorMessage.Substring(indexOfName + 4);
+
+            int indexOfExceptionType = errorMessage.IndexOf(':');
+            string exceptionType = errorMessage.Substring(0, indexOfExceptionType - 1);
+
+            XElement errorElement = new XElement("error");
+            errorElement.SetAttributeValue("name", name);
+
+            errorElement.Add(CreateFailureElement(exceptionType, errorMessage, result.ErrorStackTrace));
+
+            return errorElement;
+        }
+
+        private static XElement CreateFailureElement(string exceptionType, string message, string stackTrace)
+        {
+            XElement failureElement = new XElement("failure", new XAttribute("exception-type", exceptionType));
+            failureElement.Add(new XElement("message", RemoveInvalidXmlChar(message)));
+            failureElement.Add(new XElement("stack-trace", RemoveInvalidXmlChar(stackTrace)));
+
+            return failureElement;
+        }
+
+        private static (XElement element, int total, int passed, int failed, int skipped, int error, TimeSpan time) CreateTestSuite(IGrouping<string, TestResultInfo> resultsByType
+        )
+        {
+            var element = new XElement("test-suite");
+
+            int total = 0;
+            int passed = 0;
+            int failed = 0;
+            int skipped = 0;
+            int error = 0;
+            var time = TimeSpan.Zero;
+
+            foreach (var result in resultsByType)
+            {
+                switch (result.Outcome)
+                {
+                    case TestOutcome.Failed:
+                        failed++;
+                        break;
+
+                    case TestOutcome.Passed:
+                        passed++;
+                        break;
+
+                    case TestOutcome.Skipped:
+                    case TestOutcome.None:
+                        skipped++;
+                        break;
+                }
+
+                total++;
+                time += result.Time;
+
+                element.Add(CreateTestElement(result));
+            }
+
+            var name = resultsByType.Key;
+            var idx = name.LastIndexOf('.');
+            if (idx != -1)
+            {
+                name = name.Substring(idx + 1);
+            }
+            element.SetAttributeValue("type", "TestFixture");
+            element.SetAttributeValue("name", name);
+            element.SetAttributeValue("fullname", resultsByType.Key);
+
+            var resultString = DetermineResultFromDescendants(element, "test-case");
```

Comment:
Can we use `failed` count to get `resultstring` ?

Reply:
Yes, changed result string be determined from failed > 0 or not